bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1005,10 +1005,20 @@ private static boolean checkIsArrayType(BType sourceType, BArrayType targetType,
             if (tupleTypes.isEmpty()) {
                 return targetType.getState() == ArrayState.UNSEALED || targetType.getSize() == 0;
             }
+
+            if (sourceTupleType.getRestType() != null && targetType.getState() == ArrayState.UNSEALED &&
+                    targetType.getElementType().getTag() != TypeTags.UNION_TAG) {
```

Comment:
Why are we special casing union type?

Also instead of checking equality of type tag can't we call `checkIsType`?